README.md

```diff
@@ -373,11 +373,13 @@ following:
 Compiler and linker options
 ---------------------------
 
-CMake lets you specify a few variables to control the behavior of the
-compiler and linker. These flags are rarely used but can be useful for
-64-bit to 32-bit cross-compilation.
+There are several options that control the behavior of the compiler and
+linker. These flags may be useful for cross-compilation or specialized
+setups.
 
 - `CMAKE_C_FLAGS`: Set your own compiler flags
+- `C_STANDARD`: the C standard to compile against; defaults to `C90`
+- `C_EXTENSIONS`: whether compiler extensions are supported; defaults to `OFF`
```

Comment:
Is there a particular reason to not leverage the standard cmake variables?
https://cmake.org/cmake/help/latest/variable/CMAKE_C_STANDARD.html
https://cmake.org/cmake/help/latest/variable/CMAKE_C_EXTENSIONS.html

Reply:
😔 Not knowing about it, mostly. Does this solve the problem in a superior way?

Reply:
Well, "standard" means it is already known and/or easy to discover.
It is referenced in the documentation of the target properties, for determining the default value.
It would nicely complement the use of other standard variables (`CMAKE_C_FLAGS`).

And the standard pattern to provide default values for an input variable is to use cache variables and options. This can be discovered by CMake GUIs.
~~~
set(CMAKE_C_STANDARD "90" CACHE STRING "The C standard to compile against")
option(CMAKE_C_EXTENSIONS "Whether compiler extensions are supported" "OFF")
~~~

Reply:
Sure I understand what you're saying, but does it actually work for us given that we have deps that may have a variety of other cstd compliance levels? In other words does it apply to the whole of the system?

With this PR, we apply the cstd level just to our projects. I guess we could use this standard feature as the default and then make the deps specify their compliance levels individually. Or, I suppose, just let them error.

Reply:
> does it apply to the whole of the system?

Yes.

> does it actually work for us given that we have deps that may have a variety of other cstd compliance levels?

I can't answer that. I arrived here from package manager which de-vendors deps, so they are isolated from this setting.

But the reason I did ask is similar, only the other way: libgit2 might also be used a subproject, inheriting top-level settings.

So no change requested.

FTR the vcpkg port carries a patch to set the standard to 99, with the comment "for 'inline' in system headers". The patch can go away with the new change. But `90` isn't the right standard when it comes to Android (#6574). Maybe what I really want is libgit2 to generally chose `99` for Android instead of requiring users to make that override.

Reply:
I think that your advice is good, see #6914.

Do you think that we should default to 99 on Android? 🤔

Reply:
Why not?  
At least this is the current patch in vcpkg, building in CI with NDK r26, API level 21.  

Even if there were Android users with another toolchain: would they be hurt with the default being 99?

Reply:
@dg0yt can you test #6914 ? I don't have an android environment to test on.